pkg/reconciler/instances/istio/reset/data/gatherer.go

```diff
@@ -56,12 +56,19 @@ func (i *DefaultGatherer) GetPodsWithDifferentImage(inputPodsList v1.PodList, im
 	outputPodsList.Items = []v1.Pod{}
 
 	for _, pod := range inputPodsList.Items {
+		isIstioPod := false
```

Comment:
There could be two improvements  in this piece of code.
1. Instead of creating a new var you could just fail-fast with continue when this check is evaluated to false,
2. this logic could be wrapped into a separate private function for readability